Create_Volume_From_Snapshot.md

```diff
@@ -0,0 +1,84 @@
+Create Volumes from Snapshots
+
+**Status**: Pending
+
+**Version**: Alpha
+
+**Author(s)**: Xiaoyan Li (@lixiaoy1)
+
+## Summary
+
+This proposal is drafted for creating a volume from snapshot supported backend storages. It defines the APIinterfaces to OpenSDS customers and meanwhile the interfaces for kinds of storage drivers.
+
+## Motivation
+
+Under some scenarios users may want to reuse the data on snapshots, with this function they can easily do it.
+
+### Goals
+
+* Design a unified API interface to create volumes from snapshots supported by backend storages.
+* Complete the common logic to create volumes from snapshots, and implementations in LVM, Ceph and Cinder.
+
+### Non-Goals
+
+* The implementations inside storage drivers except LVM, Ceph and Cinder.
+* It doesnot provide the function to create volumes from snapshots which isnot supported by backend storages, like to create volumes from snapshots in different backend storages.
+
+## Proposal
+
+The proposal is drafted to allow users to create volumes from snapshots.
+
+### Data model impact
+
+VolumeSpec will be added a new item snapshot_id.
+
+type VolumeSpec struct {
+        ... // Keep the original entries.
+        SnapshotId string `json:"snapshotId,omitempty"`
+}
+
+
+### REST API impact
+
+The create-volume API URL needs to add snapshot_id parameter.
+
+### Security impact
+
+None.
+
+### Other end user impact
+
+The output of displaying volumes will add a field snapshot_id.
+
+### Performance impact
+
+None.
+
+### Other deployer impact
+
+None.
+
+### Developer impact
+
+None.
```

Comment:
Driver developers need to implement create volume from snapshot in their drivers.

Reply:
Done.